daffodil-core/src/main/scala/org/apache/daffodil/dpath/Expression.scala

```diff
@@ -1695,7 +1695,12 @@ case class FunctionCallExpression(functionQNameString: String, expressions: List
           case erd: ElementRuntimeData => erd
           case _ => SDE("dfdlx:outputTypeCalcNextSibling can only be defined on an element")
         }
-        val resolver = erd.nextElementResolver
+        val resolver = erd.partialNextElementResolver
+        //
+        // FIXME: This is broken because it is looking outward and upward for enclosing contexts
```

Comment:
remove comments. No longer relevant. The partial resolver is not looking beyond the scope of the current lexically enclosing group. 

This could be a problem if there is a group-ref/def boundary between the "siblings" in this typeCalc pair.